android/src/main/java/com/tailscale/ipn/AppSourceChecker.kt

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+package com.tailscale.ipn
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+
+object AppSourceChecker {
+
+  const val TAG = "AppSourceChecker"
+
+  fun getInstallSource(context: Context): String {
+    val packageManager = context.packageManager
+    val packageName = context.packageName
+    Log.d(TAG, "Package name: $packageName")
+
+    val installerPackageName = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        packageManager.getInstallSourceInfo(packageName).installingPackageName
+    } else {
+        packageManager.getInstallerPackageName(packageName)
+    }
+
+    Log.d(TAG, "Installer package name: $installerPackageName")
+
+    return when (installerPackageName) {
+        "com.android.vending" -> "googleplay"
+        "org.fdroid.fdroid" -> "fdroid"
+        "com.amazon.venezia" -> "amazon"
+        null -> "sideloaded"
```

Comment:
how about "unknown" instead of "sideloaded"? IIUC, "null" is also used by Samsung or other Android variants that don't implement this.

So we don't know 100% that it's sideloaded.